docs/workflows/best-practices-architecture.md

```diff
@@ -0,0 +1,76 @@
+
+<sig-toc></sig-toc>
+
+## Why modularize your codebase?
+
+Every non-trivial system is split into components. However, there are multiple different reasons for why people decide to modularize their system. Some of these reasons strictly relate to technical benefits, and other reasons purely relate to making life easier for the people working on the system. 
+
+- **Deploying components separately gives more runtime flexibility.** Runtime load is rarely spread equally across all components in the system. Deploying components separately allows for more flexibility in the deployment architecture, which is generally beneficial when it comes to managing performance behavior, scalability, and resilience. 
+- **Deploying components independently increases development flexibility.** If components are deployed separately, teams can work on those components independently and deploy 'their' component without having to simultaneously deploy the other components. However, it is quite rare for components within a system to operate in complete isolation from each other. Components tend to have various dependencies to other components. This can take the form of an interface dependency, but something like a shared database also acts as a dependency between components. The decision to deploy components separately tends to make those components a bit more independent, but not fully independent. 
+- **Dividing your codebase into logical "parts" makes it easier for people to locate functionality.** This is the most common reason to modularize systems and is done purely for the benefit of the people working on the system. Dividing a system into components helps to manage the cognitive load: People can first think about the system in terms of its high-level architecture, without being confronted with the full implementation details for every single part. The component structure is also helpful to communicate where certain functionality is implemented. 
+
+Many systems use a combination of these reasons when designing their component architecture. For example, it is quite common to define the top-level architecture as a number of microservices that are deployed separately. But every microservice also has an internal architecture, which is mainly based on dividing the codebase into logical areas. Such a combination is not "wrong". It is a perfectly reasonable choice which indicates that modularizing a system needs to consider both technical and human aspects.
+
+These considerations will change over time. That is perfectly normal and healthy. One of your component architecture's goals is to facilitate architecture evolution. It is not humanly possible to foresee all possible future changes, so it's normal for your architecture to change over time in order to meet those changing needs. That means designing your component architecture is not a one-off activity, like many other software engineering activities it's an iterative process.
```

Comment:
* for me it is ok to drop 'humanly' (also AI cannot do that ;-) )
* I have learned that you should not write abbrevs such as "it's" --> use "it is" instead (2x in this paragraph)
*